From 225c4940791647ebba9c62be46479a562047539d Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:22 -0400
Subject: [infiniband] mlx4: Fix a sparse endianness warning

Message-id: <00c1afbb1777ab2fd67666be5c341889daa61028.1410425016.git.avadai@redhat.com>
Patchwork-id: 93619
O-Subject: [Patch RHEL7.1 V1 086/197] IB/mlx4: Fix a sparse endianness warning
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Bart Van Assche <bvanassche@acm.org>

Fix the following warning for the mlx4 driver:

    $ make M=drivers/infiniband C=2 CF=-D__CHECK_ENDIAN__
    drivers/infiniband/hw/mlx4/qp.c:1885:31: warning: restricted __be16 degrades to integer

Signed-off-by: Bart Van Assche <bvanassche@acm.org>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 0e9855dbf43a9cd31df5f5f61a18e031ac1c4a82)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/mlx4/qp.c b/drivers/infiniband/hw/mlx4/qp.c
index aadf7f8..41308af 100644
--- a/drivers/infiniband/hw/mlx4/qp.c
+++ b/drivers/infiniband/hw/mlx4/qp.c
@@ -2082,7 +2082,7 @@ static int build_mlx_header(struct mlx4_ib_sqp *sqp, struct ib_send_wr *wr,
     return err;
   }
 
-  if (ah->av.eth.vlan != 0xffff) {
+  if (ah->av.eth.vlan != cpu_to_be16(0xffff)) {
    vlan = be16_to_cpu(ah->av.eth.vlan) & 0x0fff;
    is_vlan = 1;
   }
-- 
1.7.1